public/js/index.js

```diff
@@ -147,10 +147,11 @@
     });
   });
 
-  $('form.donation-variable').on('submit', function(event) {
+  $('button.donation-variable').on('click', function(event) {
     event.preventDefault();
+    var productId = $(this).data('product-id');
```

Comment:
I didn't know about the latter. It used to only work camel-cased. Maybe it now supports both for backwards compatibility.

If you want you could just get the attribute with the hyphenated attribute.